packages/calypso-e2e/src/jest-playwright-config/environment.ts

```diff
@@ -271,6 +276,9 @@ class JestEnvironmentPlaywright extends NodeEnvironment {
 			case 'test_fn_failure': {
 				this.failure = { type: 'test', name: event.test.name };
 				this.allure?.failTestStep( event.error );
+				// Print path to captured artifacts for faster triaging.
+				console.log( `Artifacts for failed test "${ event.test.name }":` );
+				console.log( this.testArtifactsPath );
```

Comment:
**suggestion:blocking**: I like the message and the you got the right idea here! 

However, placing it under `test_fn_failure` means it will trigger for each failure if the spec in question is parametrized, for instance. You can see an example of such test in [this spec](https://github.com/Automattic/wp-calypso/blob/c0567f1c501b22241ccfc1334a720e94583415f8/test/e2e/specs/infrastructure/infrastructure__ssr-enabled.ts).

The outcome of an intentionally-introduced failure for such spec file will look like this:
```
Artifacts for failed test "Check endpoint: https://wordpress.com/log-in":
/Users/edwintakahashi/workspace/wp-calypso-second/test/e2e/results/infrastructure__ssr-enabled__2022-09-29_20-19-59-756-cnHc3s
Artifacts for failed test "Check endpoint: https://wordpress.com/themes":
/Users/edwintakahashi/workspace/wp-calypso-second/test/e2e/results/infrastructure__ssr-enabled__2022-09-29_20-19-59-756-cnHc3s
Artifacts for failed test "Check endpoint: https://wordpress.com/theme/twentytwenty":
/Users/edwintakahashi/workspace/wp-calypso-second/test/e2e/results/infrastructure__ssr-enabled__2022-09-29_20-19-59-756-cnHc3s
```

Can I suggest a better place (in my opinion) for the logs, in either `teardown` or `run_finish`, so that it is only printed once?

Reply:
@worldomonation, great suggestion! I've added it to `teardown` since there was a nice `if` statement for failed tests already. It now prints once even when there are multiple failures within a spec:
`Artifacts for failed tests:`
`/Users/john/Projects/wp-calypso/test/e2e/results/fse__temp-smoke-test__2022-09-30_21-29-37-783-o4Jywy`

Unfortunately, I wasn't able to figure out hoow to print all artifact paths across **multiple** specs in one location. In other words, if I ran `yarn jest <spec_1> <spec_2>` or `yarn jest --group=calypso-pr`, the artifacts for failed tests would print after `spec_1`, then `spec_2`, etc.

Reply:
> Unfortunately, I wasn't able to figure out how to print all artifact paths across multiple specs in one location.

I think this would unfortunately be pretty complicated to do. 😕 The jest binary is spinning up multiple runners, and I don't know of hooks they provide that work at a level higher than the jest-circus events. I think what you have here looks great! 😁